packages/react-magma-dom/src/components/Input/index.tsx

```diff
@@ -54,11 +74,13 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
             descriptionId ? descriptionId : props['aria-describedby']
           }
           aria-invalid={!!errorMessage}
-          hasError={!!errorMessage}
+          hasError={!!errorMessage || characterLength > maxLength}
           iconPosition={iconPosition}
           id={id}
           inputSize={inputSize}
           isInverse={isInverse}
+          onChange={maxLength ? handleChange : props.onChange}
+          onClear={maxLength ? handleClear : props.onClear}
```

Comment:
Same with onClear. It's important to type check and ensure that `props.onClear` is actually a function before we make a call for it

```
function handleOnClear() {
   onClear && typeof onClear === 'function' && onClear();
   if (maxLength) {
      setCharacterLength(null);
   }
}
```

Also, why set the character length to `null`?